lib/createExplorer.js

```diff
@@ -74,11 +71,9 @@ module.exports = function (options) {
       })
       .then((result) => {
         if (result) return result;
-
-        const nextDirectory = joinPath(splitPath(directory).slice(0, -1));
-        if (!nextDirectory || directory === options.stopDir) return null;
-
-        return searchDirectory(nextDirectory);
+        if (!splitSearchPath) splitSearchPath = splitPath(directory);
+        if (!splitSearchPath.pop() || directory === options.stopDir) return null;
```

Comment:
Can we not use `pop()` here? I don't like how it mutates the array at the same time that we're using it to make an assertion. I'd prefer accessing the item in the array without mutation, and creating a new array without it.